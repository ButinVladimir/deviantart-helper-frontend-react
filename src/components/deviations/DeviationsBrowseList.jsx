import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeviationsPreview from './DeviationsPreview';
import Config from '../../config/config';

export default class DeviationsBrowseList extends Component {
  componentDidMount() {
    const { preloadDeviationsHandler } = this.props;
    preloadDeviationsHandler();
  }

  render() {
    const { deviations } = this.props;
    const mappedDeviations = deviations.map(d => (
      <DeviationsPreview key={d.id} {...d} />
    ));

    return (
      <ul>
        {mappedDeviations}
      </ul>
    );
  }
}

DeviationsBrowseList.propTypes = {
  config: PropTypes.instanceOf(Config).isRequired,
  deviations: PropTypes.arrayOf(PropTypes.any).isRequired,
  preloadDeviationsHandler: PropTypes.func.isRequired,
};
