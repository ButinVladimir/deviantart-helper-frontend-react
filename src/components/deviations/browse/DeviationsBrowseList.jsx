import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import DeviationsPreview from '../shared/DeviationsPreview';

export default function DeviationsBrowseList({
  deviations,
  selectedIds,
  toggleSelectionHandler,
}) {
  const mappedDeviations = deviations.map(d => (
    <DeviationsPreview
      key={d.id}
      selected={selectedIds.includes(d.id)}
      toggleSelectionHandler={toggleSelectionHandler}
      {...d}
    />
  ));

  return (
    <Section>
      <Container>
        {mappedDeviations}
      </Container>
    </Section>
  );
}


DeviationsBrowseList.propTypes = {
  deviations: PropTypes.arrayOf(PropTypes.any).isRequired,
  selectedIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  toggleSelectionHandler: PropTypes.func.isRequired,
};
