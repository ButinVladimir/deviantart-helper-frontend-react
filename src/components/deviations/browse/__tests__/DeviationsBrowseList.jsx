import React from 'react';
import { shallow } from 'enzyme';
import DeviationsBrowseList from '../DeviationsBrowseList';

describe('DeviationsBrowseList', () => {
  it('can be rendered correctly', () => {
    const deviations = ['1', '2', '3', '4'].map(id => ({
      id,
      title: `Deviation ${id}`,
      url: `/path/to/deviation/${id}`,
      preview: {
        height: 1,
        width: 2,
        src: `/path/to/thumbnail/${id}`,
      },
      views: 1,
      favourites: 2,
      comments: 3,
      downloads: 4,
    }));

    const wrapper = shallow(
      <DeviationsBrowseList
        deviations={deviations}
        selectedIds={['2', '3']}
        toggleSelectionHandler={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
