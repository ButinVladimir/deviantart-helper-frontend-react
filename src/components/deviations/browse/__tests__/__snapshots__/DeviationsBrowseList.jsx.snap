// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsBrowseList can be rendered correctly 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <DeviationsPreview
      comments={3}
      downloads={4}
      favourites={2}
      id="1"
      key="1"
      preview={
        Object {
          "height": 1,
          "src": "/path/to/thumbnail/1",
          "width": 2,
        }
      }
      selected={false}
      title="Deviation 1"
      toggleSelectionHandler={[Function]}
      url="/path/to/deviation/1"
      views={1}
    />
    <DeviationsPreview
      comments={3}
      downloads={4}
      favourites={2}
      id="2"
      key="2"
      preview={
        Object {
          "height": 1,
          "src": "/path/to/thumbnail/2",
          "width": 2,
        }
      }
      selected={true}
      title="Deviation 2"
      toggleSelectionHandler={[Function]}
      url="/path/to/deviation/2"
      views={1}
    />
    <DeviationsPreview
      comments={3}
      downloads={4}
      favourites={2}
      id="3"
      key="3"
      preview={
        Object {
          "height": 1,
          "src": "/path/to/thumbnail/3",
          "width": 2,
        }
      }
      selected={true}
      title="Deviation 3"
      toggleSelectionHandler={[Function]}
      url="/path/to/deviation/3"
      views={1}
    />
    <DeviationsPreview
      comments={3}
      downloads={4}
      favourites={2}
      id="4"
      key="4"
      preview={
        Object {
          "height": 1,
          "src": "/path/to/thumbnail/4",
          "width": 2,
        }
      }
      selected={false}
      title="Deviation 4"
      toggleSelectionHandler={[Function]}
      url="/path/to/deviation/4"
      views={1}
    />
  </Container>
</Section>
`;
